instrumentation/grpc/lib/opentelemetry/instrumentation/grpc/patches/rpc_desc.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Instrumentation
+    module GRPC
+      module Patches
+        # Patches to prepend to ::GRPC::RpcDesc.
+        module RpcDesc
+          # This is the entry point to all server-side calls.
+          def run_server_method(active_call, mth, inter_ctx = ::GRPC::InterceptionContext.new)
+            attrs = {
+              'rpc.system' => 'grpc',
+              'rpc.service' => mth.owner.service_name,
+              'rpc.method' => mth.name.to_s
+            }
+            context = OpenTelemetry.propagation.extract(active_call.metadata)
+            span = tracer.start_span(
+              "#{mth.owner.service_name}/#{mth.name}",
+              with_parent: context,
+              kind: :server,
+              attributes: attrs
+            )
+            # The server's ActiveCall is created during the accept loop, and so the
+            # current_span saved off when created will likely be a non-recording API
+            # span. So we replace the saved-off span with whatever we just created.
+            active_call.span = span
```

Comment:
Something I do not quite grok is why this does not rely on `current_span` since it is passed in to `with_span` on L32?

Reply:
You know - at this point it's been so long since I wrote it that I don't precisely recall why. I suspect it's something like what you were talking about in another area - I wanted to make sure we're augmenting the correct span, so the stuff inside the `ActiveCall` patches all operate on the `@span` variable instead of `current_span`. But; there are places (the c extension, etc) where getting at the current `ActiveCall` isn't possible so I then need to use `current_span` ... and it gets to be a mess.

Like I said elsewhere, I'm going to try harder to encapsulate "our spans" more cleanly, and then as part of that refactor I'll try to make all of this stuff a lot more clear somehow.

Reply:
We should also look at other SDK implementations as well and see if the generated code looks similar to see how they are handling it. That may help us puzzle through some options.